helpers/clean-kibana-indexes.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+# Description: script to remove old .kibana* indexes' aliases created after Kibana
+# restart.
+#
+# The aim of the script is to keep only the latest index with the latest 
+# configuration, and delete the old ones.
+
+# Usage is: 
+# CLUSTER="amazeeio-test" ELASTICSEARCH_NAMESPACE="elasticsearch" ./clean-kibana-indexes.sh"
+# CLUSTER="amazeeio-test" ELASTICSEARCH_NAMESPACE="elasticsearch" TENANT="drupalexample" ./clean-kibana-indexes.sh"
+
+set -eu
+
+# Set DEBUG variable to true, to start bash in debug mode
+DEBUG="${DEBUG:-"false"}"
+if [ "$DEBUG" = "true" ]; then
+        set -x
+fi
+
+# Cluster to connect to
+CLUSTER="${CLUSTER:-""}"
+if [ -z "$CLUSTER" ]; then
+	echo -e "Set CLUSTER variable\n"
+	echo -e "Usage is: CLUSTER="amazeeio-test" ELASTICSEARCH_NAMESPACE="elasticsearch" ./clean-kibana-indexes.sh"
+	exit 1
+fi
+
+# Set the ACTION of the script between CLOSE or DELETE
+ACTION="${ACTION:-""}"
+if [ -z "$ACTION" ]; then
+	echo -e "Set ACTION variable to CLOSE or DELETE\n"
+	echo -e "Usage is: CLUSTER="amazeeio-test" ELASTICSEARCH_NAMESPACE="elasticsearch" ACTION=CLOSE ./clean-kibana-indexes.sh"
+	exit 1
+fi
+
+# Set DRYRUN option to "true" to run in a dry-run mode
+DRYRUN="${DRYRUN:-"false"}"
+
+# Optional set a tenant to check and close/delete indexes
+TENANT="${TENANT:-"*"}"
+if [ "$TENANT" = "*" ]; then
+	REGEX="*"
+# This is a special tenant that requires a special REGEX
+elif [ "$TENANT" = "kibana" ]; then
+	REGEX=""
+else
+	REGEX="_*_$TENANT"
+fi
+
+# Switch context to the selected cluster
+echo -e "You're switching context to ${CLUSTER}"
+kubectl config use-context "${CLUSTER}"
+
+# Elasticsearch namespace of the cluster
+ELASTICSEARCH_NAMESPACE="${ELASTICSEARCH_NAMESPACE:-""}"
+
+if [ -z "$ELASTICSEARCH_NAMESPACE" ]; then
+	echo -e "\nSet ELASTICSEARCH_NAMESPACE variable"
+	echo -e "Usage is: CLUSTER="amazeeio-test" ELASTICSEARCH_NAMESPACE="elasticsearch" ./clean-kibana-indexes.sh\n"
+	exit 1
+fi
+
+# Elasticsearch POD where run the script
+ELASTICSEARCH_POD="${ELASTICSEARCH_POD:-$(kubectl -n "$ELASTICSEARCH_NAMESPACE" get pods --no-headers -o custom-columns=NAME:.metadata.name|head -n1)}"
+echo -e "Selected POD to run commands is $ELASTICSEARCH_POD"
+
+# Generate script dynamically and pipe to the selected POD
+cat << EOF |
+#/bin/bash
+IFS=$'\n'
+
+# Generate array with kibana aliases for a specific tenant if specified, otherwise get all aliases
+ELASTICSEARCH_ALIASES=(\$(es-curl GET "_cat/aliases/.kibana$REGEX?h=alias,index"))
```

Comment:
don't you need to escape the `$` here? we do it everywhere else

Reply:
This is how I can define a variable outside the script generated by EOF and pass it to it.
The escaped `$` is because defined variables are used inside the script itself, while REGEX, for example, is defined outside.